backend/database.py

```diff
@@ -1,26 +1,226 @@
+    print(result)
+    return result
+
+def get_all_classes(c):
+    command = '''
+    SELECT * FROM Classes
+    '''
+    cur = c.cursor()
+    cur.execute(command)
+    result = cur.fetchall()
+    print(result)
+    return result
+
+def get_all_users(c):
+    command = '''
+    SELECT * FROM Users
+    '''
+    cur = c.cursor()
+    cur.execute(command)
+    result = cur.fetchall()
+    print(result)
+    return result
+
+def get_user(user, c):
+    command = '''
+    SELECT * FROM Users WHERE u_name =''' + "\""+ user +"\"" +";"
+    cur = c.cursor()
+    cur.execute(command)
+    result = cur.fetchone()
+    return result
+
+def get_classes_of_student(c, user):
+    command = '''
+    SELECT c_classname, c_teacher, c_time, c_seats
+    FROM Classes, Students_in_Classes
+    WHERE sc_name = \"''' + user + '''\"
+    and sc_classkey = c_classkey
+    '''
+    cur = c.cursor()
+    cur.execute(command)
+    result = cur.fetchall()
+    print(result)
+    return result
+
+def get_classes_of_teacher(c, user):
+    command = '''
+    SELECT c_classkey, c_classname, c_teacher, c_time, c_seats
+    FROM Classes
+    WHERE c_teacher = \"''' + user + '''\"'''
+    cur = c.cursor()
+    cur.execute(command)
+    result = cur.fetchall()
+    print(result)
+    return result
+
+def get_students_in_classes(c):
+    command = '''
+    SELECT * FROM Students_in_CLasses
+    '''
+    cur = c.cursor()
+    cur.execute(command)
+    result = cur.fetchall()
+    print(result)
+    return result
+
+def get_students_in_a_class(classkey, c):
+    command = '''
+    SELECT sc_name, sc_grade
+    FROM Students_in_Classes
+    WHERE sc_classkey = ''' + str(classkey)
+    cur = c.cursor()
+    cur.execute(command)
+    result = cur.fetchall()
+    print(result)
+    return result
+
+def create_db(c):
+    command = '''
+    CREATE TABLE Users (
+        u_name varchar(255) primary key not null,
+        u_password varchar(255) not null,
+        u_role varchar(1) not null
+    );
+    '''
+    c.execute(command)
+
+    command = '''
+    CREATE TABLE Classes (
+        c_classkey int primary key not null,
+        c_classname varchar(255) DEFAULT "TBA" not null,
+        c_teacher varchar(255) DEFAULT "TBA" not null,
+        c_time varchar(255) DEFAULT "TBA" not null,
+        c_seats int not null
+    );   
+    '''
+    c.execute(command)
+
+    command = '''
+    CREATE TABLE Students_in_Classes (
+        sc_classkey int not null,
+        sc_name varchar(255) not null,
+        sc_grade int not null,
+        primary key (sc_classkey, sc_name)
+    )
+    '''
+    c.execute(command)
+    c.commit()
+
+def insert_user(name, password, type, c):
+    command = '''
+    INSERT INTO Users
+    values('''+\
+    "\""+str(name)+"\", "+\
+    "\""+str(password)+"\", "+\
+    "\""+str(type)+"\");"
+    
+    c.execute(command)
+    c.commit()
+
+def delete_user(name, c):
+    get_result = get_user(name, c)
+
+    if get_result != None:
+        user_type = get_result[2]
```

Comment:
We should probably be using "roles" rather than "types". Also, roles should be single words.